webviz_subsurface/_datainput/eclipse_init_io/pvt_common.py

```diff
@@ -0,0 +1,872 @@
+    """A base class for dead and dry gas/oil respectively.
+
+    Attributes:
+        x: A numpy array holding the independent values.
+        y: A two-dimensional numpy array holding the dependent values.
+        interpolation: A scipy interp1d object for interpolating the dependent values.
+        inter_extrapolation: An extrap1d object for inter- and extrapolating the dependent values.
+    """
+
+    def __init__(
+        self,
+        index_table: int,
+        raw: EclPropertyTableRawData,
+        convert: ConvertUnits,
+    ) -> None:
+        # pylint: disable=super-init-not-called
+        """Extracts all values of the table with the given index from raw, converts them according
+        to the given convert object and stores them as numpy arrays, x and y respectively.
+
+        Creates an interpolation and an extrapolation object utilising scipy's interp1d and based on
+        it the custom tailored extrap1d.
+
+        Raises a ValueError if there is no interpolation interval given, that is when there are
+        fewer than two independents.
+
+        Args:
+            index_table: The index of the table which values are supposed to be extracted.
+            raw:
+                An EclPropertyTableRawData object that was initialised based on an Eclipse
+                INIT file.
+            convert: A ConvertUnit object that contains callables for converting units.
+
+        """
+        self.x: np.ndarray
+        self.y: np.ndarray = np.zeros((0, 0))
+
+        column_stride = raw.num_rows * raw.num_tables * raw.num_primary
+        table_stride = raw.num_primary * raw.num_rows
+
+        for index_primary in range(0, raw.num_primary):
+            if self.entry_valid(raw.primary_key[index_primary]):
+                for index_row in range(0, raw.num_rows):
+                    current_stride = (
+                        index_table * table_stride
+                        + index_primary * raw.num_rows
+                        + index_row
+                    )
+
+                    if self.entry_valid(raw.data[column_stride * 0 + current_stride]):
+                        self.x = np.append(
+                            self.x,
+                            convert.independent(
+                                raw.data[column_stride * 0 + current_stride]
+                            ),
+                        )
+
+                        for index_column in range(1, raw.num_cols):
+                            self.y[index_column] = np.append(
+                                self.y[index_column],
+                                convert.column[index_column - 1](
+                                    raw.data[
+                                        column_stride * index_column + current_stride
+                                    ]
+                                ),
+                            )
+
+                    else:
+                        break
+            else:
+                break
```

Comment:
Ran into a bug here. You are both appending to `self.x` prior to initializing it the first time, and also indexing `self.y` which is initialized as an array of shape (0,0). 

Also: Appending to an array makes a copy with appended data. I think it would be possible here to precalculate the dimensions of `x` and `y`, and preallocate arrays of the right dimension?